client-side/src/components/Report/report.jsx

```diff
@@ -0,0 +1,126 @@
+import { useEffect, useState } from "react";
+import axios from 'axios';
+import Select from '../../stories/Select/Select.jsx';
+import TableComponent from '../../stories/table/TableComponent';
+import Loader from "../../stories/loader/loader";
+import GenericButton from '../../stories/Button/GenericButton';
+import DateInput from "../../stories/DateTime/DateInput";
+import PdfGenerator from "./pdf.jsx"
+import { OPTION_ARRAY, TIME } from './report.constant.jsx';
+import './report.scss'
+
+export default function Report() {
+
+  const [data, setData] = useState([])
+  const [customArr, setCustomArr] = useState([])
+  const [selectType, setSelectType] = useState(TIME.Month.text);
+  const [startDate, setStartDate] = useState('');
+  const [endDate, setEndDate] = useState('');
+  const [dateTimePiker, setDateTimePiker] = useState(false);
+
+  const selectFunction = (selectedValue) => {
+    switch (selectedValue) {
+      case 1:
+        setSelectType(TIME.DAY.text);
+        setDateTimePiker(false);
+        break;
+      case 2:
+        setSelectType(TIME.Month.text);
+        setDateTimePiker(false);
+        break;
+      case 3:
+        setSelectType(TIME.YEAR.text);
+        setDateTimePiker(false);
+        break;
+      case 4:
+        setSelectType(TIME.CUSTOM.text);
+        setDateTimePiker(true);
+        break;
+      default:
+        break;
+    }
+    if (selectedValue != 4) {
+      fillData();
+    }
+  };
+
+  useEffect(() => {
+    fillData();
+  }, [data])
+
+
+  const storeArr = () => {
+    setCustomArr([startDate.$d.toISOString(), endDate.$d.toISOString()]);
+    fillData();
+  }
+
+  const isButtonDisabled = (!startDate || !endDate) || startDate > endDate;
+
+  const handleDateChange = (inputName, date) => {
+    if (inputName === 'start') {
+      setStartDate(date);
+    } else {
+      setEndDate(date);
+    }
+  }
+
+
+
+  const fillData = async () => {
+
+    const body = {
+      userId: "66a21c983652170a94fe4247",
+      type: selectType,
+      customDates: selectType === "custom" ? customArr : null
+    }
+
+    try {
+      await axios.post(`http://localhost:5000/vistedWebsite/showVisitedWebsite`, body)
+        .then(res => {
+          setData(res.data)
+        })
+    }
+    catch {
+      console.log("faild")
+    }
+
+  }
+
+  let args = {
+    dataObject: {
+      headers: ['id', 'website_name', 'total_time', 'avg_for_day'],
```

Comment:
please replace with readable titles, for example "Website Name" instead of "website_name"
move titles to constants file